### PR TITLE
One-click upsell A/B test for Personal to Premium upgrade

### PR DESCRIPTION
**Pull request by niranjan-uma-shankar**
#### Changes proposed in this Pull Request

* Implement an A/B test for Personal plan purchases:
   - Control group will see a Quick Start offer after purchase completes for a Personal plan
    - Treatment group will see a offer for Premium plan upgrade with one-click purchase enabled.

* For more context, please read pcbrnV-LX-p2.

**Notes before you begin review**

This PR looks bigger than the actual code changes because I've moved some existing files to new folders, so those appear as new code changes. Here's a quick summary of some of those changes for easier review:

* Renamed the folder `client/my-sites/checkout/upsell-nudge/plan-upgrade-upsell` to `client/my-sites/checkout/upsell-nudge/business-plan-upgrade-upsell/`
* Created a new folder `client/my-sites/checkout/upsell-nudge/premium-plan-upgrade-upsell/` that has the Premium upsell offer page and its related sass file.
* Important: The copy for the Premium upgrade offer is right now a placeholder content and not the final version. It will be replaced with the actual copy whenever it's ready. 

#### Testing instructions

1. Assign yourself to the treatment of `one_click_premium_plan_upgrade` A/B test and to `offer` variation of `conciergeUpsellDial` test.
2. Signup for a new WPCOM account and purchase a Personal plan
3. Verify that you see the Premium plan upsell offer.
4. Start over but this time assign to control of `one_click_premium_plan_upgrade` A/B test. Repeat steps 1-3 and verify that you see  the Quick Start offer.

### CODE REVIEW

File: client/my-sites/checkout/upsell-nudge/premium-plan-upgrade-upsell/index.jsx [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+							</b>
+						</p>
+						<p>
+							{ translate(
+								"Wouldn't you like to make a great first impression in those three seconds?"
+							) }
+						</p>
+						<p>
+							{ translate(
+								'That’s exactly why we’ve partnered with some of the world’s greatest designers to offer nearly 200 high-end designs that you can use to make your site looks incredible.'
```
Comment (a8ci18n) [inline client/my-sites/checkout/upsell-nudge/premium-plan-upgrade-upsell/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'That's exactly why we've partnered with some of the world's greatest designers to offer high-end designs that you can use to make your site look incredible.' )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/1-click-premium-bump)

File: client/my-sites/checkout/upsell-nudge/premium-plan-upgrade-upsell/index.jsx [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+							) }
+						</p>
+						<p>
+							{ translate(
+								'You’ll also gain access to some of the most powerful features on WordPress.com:'
```
Comment (a8ci18n) [inline client/my-sites/checkout/upsell-nudge/premium-plan-upgrade-upsell/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'You'll also gain access to some of the most powerful features on WordPress.com:' )` *ES Score: 8*

File: client/my-sites/checkout/upsell-nudge/premium-plan-upgrade-upsell/index.jsx [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+								translate(
+									'Give the Premium plan a risk-free test drive with our {{b}}%(days)d-day money-back guarantee{{/b}}.',
+									{
+										args: { days: 7 },
+										components: { b: <b /> },
+									}
+								) }
+							{ ! hasSevenDayRefundPeriod &&
+								translate(
+									'Give the Premium plan a risk-free test drive with our {{b}}30-day money-back guarantee{{/b}}.',
```
Comment (a8ci18n) [inline client/my-sites/checkout/upsell-nudge/premium-plan-upgrade-upsell/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Give the Premium plan a risk-free test drive with our {{u}}30-day Money Back Guarantee{{/u}}.' )` *ES Score: 8*

File: client/my-sites/checkout/upsell-nudge/premium-plan-upgrade-upsell/index.jsx [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+									{
+										components: { b: <b /> },
+									}
+								) }
+						</p>
+						<p>
+							{ hasSevenDayRefundPeriod &&
+								translate(
+									'Simply click the link below and select the Premium plan option to upgrade today {{b}}for just {{del}}%(fullPrice)s{{/del}} %(discountPrice)s more{{/b}}. Once you upgrade, you’ll have %(days)d days to evaluate the plan and decide if it’s right for you.',
```
Comment (a8ci18n) [inline client/my-sites/checkout/upsell-nudge/premium-plan-upgrade-upsell/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Simply click the link below and select the Business plan option to upgrade today {{b}}for just {{del}}%(fullPrice)s{{/del}} %(discountPrice)s more{{/b}}. Once you upgrade, you’ll have %(days)d days to evaluate the plan and decide if it’s right for you.' )` *ES Score: 9*

File: client/my-sites/checkout/upsell-nudge/premium-plan-upgrade-upsell/index.jsx [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+											fullPrice: formatCurrency( planRawPrice, currencyCode, { stripZeros: true } ),
+											discountPrice: formatCurrency( planDiscountedRawPrice, currencyCode, {
+												stripZeros: true,
+											} ),
+										},
+									}
+								) }
+							{ ! hasSevenDayRefundPeriod &&
+								translate(
+									'Simply click the link below and select the Premium plan option to upgrade today {{b}}for just {{del}}%(fullPrice)s{{/del}} %(discountPrice)s more{{/b}}. Once you upgrade, you’ll have 30 days to evaluate the plan and decide if it’s right for you.',
```
Comment (a8ci18n) [inline client/my-sites/checkout/upsell-nudge/premium-plan-upgrade-upsell/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Simply click the link below and select the Business plan option to upgrade today {{b}}for just {{del}}%(fullPrice)s{{/del}} %(discountPrice)s more{{/b}}. Once you upgrade, you’ll have 30 days to evaluate the plan and decide if it’s right for you.' )` *ES Score: 9*

File: client/my-sites/checkout/upsell-nudge/premium-plan-upgrade-upsell/index.jsx [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+					onClick={ handleClickDecline }
+				>
+					{ translate( 'No thanks, I’ll stick with the free themes' ) }
+				</Button>
+				<Button
+					primary
+					className="premium-plan-upgrade-upsell__accept-offer-button"
+					onClick={ () => handleClickAccept( 'accept' ) }
+				>
+					{ translate( 'Yes, I’d love to try those Premium designs!' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/upsell-nudge/premium-plan-upgrade-upsell/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Yes, I'd love to try those Premium designs!' )` *ES Score: 7*

File: client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/index.ts [hunk 7]
```diff
@@ -42,6 +43,8 @@ export default function useGetThankYouUrl( {
 	isInEditor?: boolean;
 } ): GetThankYouUrl {
 	const selectedSiteData = useSelector( ( state ) => getSelectedSite( state ) );
+	const shouldShowOneClickTreatment = useSelector( ( state ) => isTreatmentOneClickTest( state ) );
```
Comment (yanirs) [inline client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/index.ts]: @jessie-ross @withinboredom can this be simplified in light of #46805?

Comment (jessie-ross) [reply client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/index.ts]: I don't think that API is ready for use yet, see #48056 for what might be a long term fix to having a simpler Calypso client.

File: client/my-sites/checkout/composite-checkout/composite-checkout.tsx [hunk 8]
```diff
@@ -691,6 +692,7 @@ export default function CompositeCheckout( {
 			<QuerySitePlans siteId={ siteId } />
 			<QuerySitePurchases siteId={ siteId } />
 			<QueryPlans />
+			<QueryExperiments />
```
Comment (jessie-ross) [inline client/my-sites/checkout/composite-checkout/composite-checkout.tsx:575]: Ah! @niranjan-uma-shankar something not taken account of here is that experiments have a loading state.
While an experiment is loading neither control or treatment should be rendered.
Does this query happen early enough that a loading state can be safely ignored?

Comment (jessie-ross) [reply client/my-sites/checkout/composite-checkout/composite-checkout.tsx:575]: There is an `isLoading` selector if need be.

File: client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts [hunk 9]
```diff
@@ -334,6 +340,13 @@ function getRedirectUrlForConciergeNudge( {
 		// being offered and so sold, to be inline with HE availability.
 		// To dial back, uncomment the condition below and modify the test config.
 		if ( 'offer' === abtest( 'conciergeUpsellDial' ) ) {
+			if ( hasPersonalPlan( cart ) ) {
+				if ( shouldShowOneClickTreatment ) {
+					const upgradeItem = hasMonthlyCartItem( cart ) ? 'premium-monthly' : 'premium';
+					return `/checkout/${ siteSlug }/offer-plan-upgrade/${ upgradeItem }/${ pendingOrReceiptId }`;
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts:352]: Please add a test for this in https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/composite-checkout/test/composite-checkout-thank-you.js!

File: client/my-sites/checkout/composite-checkout/composite-checkout.tsx (hunk 8 above)
Comment (niranjan-uma-shankar) [reply client/my-sites/checkout/composite-checkout/composite-checkout.tsx:575]: @jessie-ross The query happens early so the loading state can be safely ignored. This query happens when the checkout page loads, and only after the user has completed purchase of a Personal plan do we determine to show the offer page based on the variation.

File: client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts (hunk 9 above)
Comment (niranjan-uma-shankar) [reply client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts:352]: Test added!

**Review by taggon — APPROVED** [review]
Works as described.
I was about to add a comment just like what @yanirs said but it would be better to not use it since @jessie-ross is working on the hooks.

File: client/my-sites/checkout/upsell-nudge/premium-plan-upgrade-upsell/index.jsx [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+								</span>
+							</li>
+							<li className="premium-plan-upgrade-upsell__checklist-item">
+								<Gridicon
+									icon="checkmark"
+									className="premium-plan-upgrade-upsell__checklist-item-icon"
+								/>
+								<span className="premium-plan-upgrade-upsell__checklist-item-text">
+									{ translate(
+										'{{strong}}Advanced social media tools{{/strong}}. Schedule posts in advance, resurface your older content, or schedule multiple social posts at a time.',
```
Comment (a8ci18n) [inline client/my-sites/checkout/upsell-nudge/premium-plan-upgrade-upsell/index.jsx:165]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( '{{b}}Advanced tools to become a social media pro.{{/b}} Schedule posts in advance, resurface your older content, or share multiple social posts at a time.' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/1-click-premium-bump)

**Pull request merged**